apps/meteor/app/api/server/v1/users.ts

```diff
@@ -261,7 +261,8 @@ API.v1.addRoute(
 				if (fields.userId) {
 					user = await Users.findOneById(fields.userId, { projection: { username: 1 } });
 				} else if (fields.username) {
-					user = await Users.findOneByUsernameIgnoringCase(fields.username, { projection: { username: 1 } });
+					// TODO: Fix these types. We're passing a projection but the `user` variable is expecting way more data than this projection
```

Comment:
The projection is actually fine. The issue here is that the variable `user` is being re-used and the two different uses have different implicit types. You can just use a new variable inside of the `sentTheUserByFormData` condition (and change the call to setUserAvatar accordingly)

Reply:
Yup, left this one for a separate PR. Didn't want to change that much on here as PR was getting bigger than expected.